docs/standard/exceptions.md

```diff
@@ -0,0 +1,528 @@
+
+Applications must be able to handle errors that occur during execution in a consistent manner. .NET provides a model for notifying applications of errors in a uniform way: .NET operations indicate failure by throwing exceptions.
+
+## Exceptions
+
+An exception is any error condition or unexpected behavior that is encountered by an executing program. Exceptions can be thrown because of a fault in your code or in code that you call (such as a shared library), unavailable operating system resources, unexpected conditions that the runtime encounters (such as code that cannot be verified), and so on. Your application can recover from some of these conditions, but not from others. Although you can recover from most application exceptions, you cannot recover from most runtime exceptions.
+
+In .NET, an exception is an object that inherits from the [System.Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) class. An exception is thrown from an area of code where a problem has occurred. The exception is passed up the stack until the application handles it or the program terminates.
+
+## Exceptions vs. traditional error-handling methods
+
+Traditionally, a language's error-handling model relied on either the language's unique way of detecting errors and locating handlers for them, or on the error-handling mechanism provided by the operating system. .NET implements exception handling with the following features:
+
+- Handles exceptions without regard for the language that generates the exception or the language that handles the exception.
+
+- Does not require any particular language syntax for handling exceptions, but allows each language to define its own syntax.
+
+- Allows exceptions to be thrown across process and even machine boundaries.
+
+Exceptions offer several advantages over other methods of error notification, such as return codes. Failures do not go unnoticed. Invalid values do not continue to propagate through the system. You do not have to check return codes. Finally, exception-handling code can be easily added to increase program reliability.
+
+## How the runtime manages exceptions
+
+The runtime uses an exception-handling model based on exception objects and protected blocks of code. An [Exception](https://docs.microsoft.com/dotnet/core/api/System.Exception) object is created to represent an exception when it occurs.
+
+The runtime creates an exception information table for each executable. Each method of the executable has an associated array of exception-handling information (which can be empty) in the exception information table. Each entry in the array describes a protected block of code, any exception filters associated with that code, and any exception handlers (**catch** statements). This exception table is extremely efficient and does not cause any performance penalty in processor time or in memory use when an exception does not occur. You use resources only when an exception occurs.
```

Comment:
I think we should limit discussion of the implementation details. Instead, we should link here for that: https://github.com/dotnet/coreclr/blob/master/Documentation/botr/exceptions.md. Instead, we should focus on the user model.

If there is an exception table, then there is memory cost. It's hard to suggest there is none.